src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,138 @@
-import { PeopleFilters } from './PeopleFilters';
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { getPeople } from '../api';
 import { Loader } from './Loader';
+import { Person } from '../types';
+import { PeopleFilters } from './PeopleFilters';
 import { PeopleTable } from './PeopleTable';
 
 export const PeoplePage = () => {
+  const [peopleFromServer, setPeopleFromServer] = useState<Person[]>([]);
+  const [error, setError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  // const [people, setPeople] = useState(peopleFromServer);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries');
+
+  // #region initialLoad
+  const getPeopleFromServer = async () => {
+    try {
+      setIsLoading(true);
+      const response = await getPeople();
+
+      setPeopleFromServer(response);
+    } catch {
+      setError(true);
+      setTimeout(() => setError(false), 1000);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getPeopleFromServer();
+  }, []);
+  // #endregion
+
+  // #region getVisiblePeople
+
+  let visiblePeople = [...peopleFromServer];
```

Comment:
Should be used a separate function to filter and sort from line 45 to line 89 which will return filtered array